local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -564,8 +564,55 @@ void LocalPlannerNode::publishWaypoints(bool hover) {
   mavros_obstacle_free_path_pub_.publish(obst_free_path);
 }
 
-void LocalPlannerNode::publishHistogramImage() {
+void LocalPlannerNode::publishDataImages() {
+  sensor_msgs::Image cost_img = local_planner_->cost_image_;
+  cost_img.header.stamp = ros::Time::now();
+
+  // current orientation
+  Eigen::Quaternionf q = toEigen(newest_pose_.pose.orientation);
```

Comment:
I believe this whole block appears in many places. I'd suggest a helper function in common 
`float getYawFromPose(const geometry_msgs::PoseStamped& pose)`

Reply:
you are right the angle situation with the two conventions is very confusing. I added the function you suggested and tried to clean up a bit at least such that the variable names reflect the coordinate system.